paddle/fluid/framework/ir/mkldnn/compute_propagate_scales_mkldnn_pass.cc

```diff
@@ -0,0 +1,427 @@
+// Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <float.h>
+#include <algorithm>
+
+#include "paddle/fluid/framework/ir/graph_helper.h"
+#include "paddle/fluid/framework/ir/mkldnn/compute_propagate_scales_mkldnn_pass.h"
+#include "paddle/fluid/framework/ir/mkldnn/mkldnn_pass_util.h"
+#include "paddle/fluid/framework/op_version_registry.h"
+
+namespace paddle {
+namespace framework {
+namespace ir {
+
+void ComputePropagateScalesMkldnnPass::GetTensorFromVector(
+    const std::vector<float>& data_v, Tensor* tensor) const {
+  const int size = static_cast<int>(data_v.size());
+  auto* data = tensor->mutable_data<float>({size}, platform::CPUPlace());
+  for (int i = 0; i < size; i++) {
+    data[i] = data_v[i];
+  }
+}
+
+void ComputePropagateScalesMkldnnPass::GetQuantInfo(
+    ir::Graph* graph, StringPairMap* var_quant_scales) const {
+  std::unordered_map<std::string, std::vector<float>> info_map{};
+  GetInfoFromTheFirstOp(graph, "has_quant_info", "var_quant_scales", &info_map);
+
+  for (auto iter = info_map.begin(); iter != info_map.end(); iter++) {
+    Tensor tensor;
+    GetTensorFromVector(iter->second, &tensor);
+    auto pair = std::make_pair(false, tensor);
+    var_quant_scales->insert(std::make_pair(iter->first, pair));
+  }
+}
+
+std::vector<float> ComputePropagateScalesMkldnnPass::GetScales(Tensor* tensor,
+                                                               int axis) const {
+  PADDLE_ENFORCE_LT(axis, 2, "The input axis is required to be less than 2.");
+  auto* data = tensor->mutable_data<float>(platform::CPUPlace());
+  const auto dims = tensor->dims();
+  PADDLE_ENFORCE_EQ(dims.size(), 2,
+                    "The input tensor's rank is required to be 2.");
+
+  const int rows = dims.at(0);
+  const int columns = dims.at(1);
+  std::vector<float> scales;
+  if (axis == 0) {
+    for (int i = 0; i < columns; i++) {
+      float max_value = FLT_MIN;
+      for (int j = 0; j < rows; j++) {
+        max_value = std::max(max_value, data[i + j * columns]);
```

Comment:
Here should be max absolute value, so I guess it should be  `std::abs(data[i + j * columns])`
This is a reference part from quant2_int8_mkldnn_pass 
https://github.com/PaddlePaddle/Paddle/blob/29d2e9497faa17382272c71d3e265ba1be373122/python/paddle/fluid/contrib/slim/quantization/quant2_int8_mkldnn_pass.py#L521

Reply:
OK~